gpt_engineer/chat_to_files.py

```diff
@@ -40,3 +41,57 @@ def to_files(chat, workspace):
     files = parse_chat(chat)
     for file_name, file_content in files:
         workspace[file_name] = file_content
+
+
+def overwrite_files(chat, dbs, replace_files):
+    """
+    Replace the AI files to the older local files.
+    """
+    dbs.workspace["all_output.txt"] = chat
+
+    files = parse_chat(chat)
+    for file_name, file_content in files:
+        # Verify if the file created by the AI agent was in the input list
+        if file_name in replace_files:
+            # If the AI created a file from our input list, we replace it.
+            with open(replace_files[file_name], "w") as text_file:
+                text_file.write(file_content)
+        else:
+            # If the AI create a new file I don't know where to put it yet
+            # maybe we can think in a smarter solution for this in the future
+            # like asking the AI where to put it.
+            #
+            # by now, just add this to the workspace inside .gpteng folder
+            print(
+                f"Could not find file path for '{file_name}', creating file in workspace"
+            )
+            dbs.workspace[file_name] = file_content
+
+
+def get_code_strings(input) -> dict[str, str]:
+    """
+    Read file_list.txt and return file names and its content.
+    """
+    files_paths = input["file_list.txt"].strip().split("\n")
+    files_dict = {}
+    for file_path in files_paths:
+        file_data = ""
+        with open(file_path, "r") as file:
+            file_data = file.read()
+        if file_data:
+            file_name = os.path.basename(file_path).split("/")[-1]
```

Comment:
We could, however we would have to communicate that whole path with the AI in our prompts.  
My human brain says that's too much information but I'm just making assumptions.  It would be worthwhile to do an experiment with sending the whole path to the AI.  I can do that.

Reply:
It works with using the whole path.  The prompt for the return file type needed to be modified as well.